screens/search/Search.tsx

```diff
@@ -38,83 +39,61 @@ const Search = () => {
-  const subMatcherData = substringMatcherQuery.data;
-
-  const activeRoutesIsLoading = activeRoutesQuery.isLoading;
-  const activeRoutesIsError = activeRoutesQuery.isError;
-  const activeRoutesError = activeRoutesQuery.error;
-  const activeRoutesData = activeRoutesQuery.data;
-
   useEffect(() => {
-    if (subMatcherData === undefined) return;
+    if (
+      userMatcherQuery.data === undefined ||
+      archivedMatcherQuery.data === undefined
+    )
+      return;
     if (query === '') {
       setArchivedRoutesSearchResults([]);
       return;
     }
     if (tabViewed === 'archived') {
       setIsLoadingResults(true);
       setArchivedRoutesSearchResults(
-        subMatcherData.archivedRoutesSubstringMatcher.getMatches(query)
+        archivedMatcherQuery.data.getMatches(query)
       );
       setIsLoadingResults(false);
     }
-  }, [query, subMatcherData, tabViewed]);
-
-  useEffect(() => {
-    if (subMatcherData === undefined) return;
-    if (query === '') {
-      setUserSearchResults([]);
-      return;
-    }
     if (tabViewed === 'users') {
       setIsLoadingResults(true);
-      setUserSearchResults(
-        subMatcherData.userSubstringMatcher.getMatches(query)
-      );
+      setUserSearchResults(userMatcherQuery.data.getMatches(query));
       setIsLoadingResults(false);
     }
-  }, [query, subMatcherData, tabViewed]);
+  }, [archivedMatcherQuery.data, query, tabViewed, userMatcherQuery.data]);
 
   useEffect(() => {
-    if (activeRoutesData === undefined) return;
+    if (activeRoutesQuery.data === undefined) return;
     if (query === '') {
       setDisplayedRouteIndices([]);
       return;
     }
     if (tabViewed === 'active') {
       setIsLoadingResults(true);
       const newDisplayedRouteIndices: number[] = [];
-      activeRoutesData.activeRoutes.forEach((route, index) => {
-        const corpus = [route.name, route.grade, route.stringifiedTags]
+      activeRoutesQuery.data.activeRoutes.forEach((route, index) => {
+        const corpus = [
+          route.name,
+          route.gradeDisplayString,
+          route.stringifiedTags,
+        ]
           .join('$')
           .toLowerCase();
         if (corpus.includes(query)) newDisplayedRouteIndices.push(index);
       });
       setDisplayedRouteIndices(newDisplayedRouteIndices);
       setIsLoadingResults(false);
     }
-  }, [query, activeRoutesData, tabViewed]);
+  }, [activeRoutesQuery.data, query, tabViewed]);
 
-  if (subMatcherIsLoading || activeRoutesIsLoading) {
-    return;
-  }
-
-  if (
-    subMatcherIsError ||
-    activeRoutesIsError ||
-    subMatcherData === undefined ||
-    activeRoutesData === undefined
-  ) {
-    console.error(activeRoutesError);
-    console.error(subMatcherError);
-    return null;
-  }
+  if (userMatcherQuery.isError) console.error(userMatcherQuery.error);
+  if (archivedMatcherQuery.isError) console.error(archivedMatcherQuery.error);
+  if (activeRoutesQuery.isError) console.error(activeRoutesQuery.error);
```

Comment:
can we return `null` on these?

Also, why aren't we blocking when the queries are loading anymore?

Reply:
Cuz it leads to a weird and unnecessary UX where you click on the search tab and it hangs for like a second or however long it takes to get the data to search on. Like I mentioned in the PR message, this doesn't need to happen, we can render the search bar and then the data fetching can block the actual searching, but the search bar should be responsive

Reply:
This is also why I decided not to return `null` on these; if for whatever reason we don't get one of the matchers, well that sucks but the other parts of the search bar should remain functional

Reply:
Why not render a spinner in the content below the search bar instead? We don't need to get rid of the whole spinner.

Reply:
I'm confused.. what spinner did I get rid of?

Reply:
Sorry, you didn't get rid of it. We never had one.

The UX just feels wrong right now, and I think the root of it is that there isn't a clear distinction between "searching" and "done searching but no results". Rendering a spinner while searching (and loading matchers), would fix this.